openlibrary/plugins/openlibrary/js/droppers/index.js

```diff
@@ -0,0 +1,46 @@
+import { debounce } from '../nonjquery_utils';
+
+/**
+ * Holds references to each dropper on a page.
+ * @type {Array<HTMLElement>}
+ */
+const droppers = []
+
+/**
+ * Adds expand and collapse functionality to our droppers.
+ *
+ * @param {HTMLCollection<HTMLElement>} dropperElements
+ */
+export function initDroppers(dropperElements) {
+    /**
+     * close an open dropdown in a given container
+     * @param {jQuery.Object} $container
+     */
+    function closeDropdown($container) {
+        $container.find('.dropdown').slideUp(25);
+        $container.find('.arrow').removeClass('up');
+    }
+
+    for (const dropper of dropperElements) {
+        droppers.push(dropper)
+
+        $(dropper).on('click', '.dropclick', debounce(function() {
+            $(this).next('.dropdown').slideToggle(25);
+            $(this).parent().next('.dropdown').slideToggle(25);
+            $(this).parent().find('.arrow').toggleClass('up');
+        }, 300, false))
+
+        $(dropper).on('click', '.dropper__close', debounce(function() {
```

Comment:
I've added the `dropper__close` class to dropper affordances that, when clicked, should close the dropper (reading log actions and adding to a list).
The concept of closing a dropper by clicking on an element with class `dropper__close` is meant to be short-lived.  I'm planning on creating a generic dropper component, and will have a solution for closing the droppers that improves the separation of concerns between a dropper and its contents.